src/core/src/OmniGlobeAnchor.cpp

```diff
@@ -0,0 +1,125 @@
+#include "cesium/omniverse/OmniGlobeAnchor.h"
+
+#include "cesium/omniverse/GeospatialUtil.h"
+#include "cesium/omniverse/UsdUtil.h"
+
+#include <glm/ext/matrix_relational.hpp>
+#include <glm/ext/matrix_transform.hpp>
+#include <glm/gtx/euler_angles.hpp>
+#include <glm/gtx/matrix_decompose.hpp>
+
+#include <utility>
+
+namespace cesium::omniverse {
+
+const pxr::GfMatrix4d OmniGlobeAnchor::getCachedTransformation() {
+    return _valueCache.transformation;
+}
+
+const pxr::GfVec3d OmniGlobeAnchor::getCachedGeographicCoordinate() {
+    return _valueCache.geographicCoordinate;
+}
+
+const pxr::GfVec3d OmniGlobeAnchor::getCachedEcefPosition() {
+    return _valueCache.ecefPosition;
+}
+
+const pxr::GfVec3d OmniGlobeAnchor::getCachedEcefRotation() {
+    return _valueCache.ecefRotation;
+}
+
+const pxr::GfVec3d OmniGlobeAnchor::getCachedEcefScale() {
+    return _valueCache.ecefScale;
+}
+
+void OmniGlobeAnchor::updateCachedValues() {
+    auto globeAnchorAPI = UsdUtil::getCesiumGlobeAnchor(_anchorPrimPath);
+
+    pxr::GfVec3d newGeographicCoordinate;
+    globeAnchorAPI.GetGeographicCoordinateAttr().Get(&newGeographicCoordinate);
+    _valueCache.geographicCoordinate = newGeographicCoordinate;
+
+    pxr::GfVec3d newEcefPosition;
+    globeAnchorAPI.GetPositionAttr().Get(&newEcefPosition);
+    _valueCache.ecefPosition = newEcefPosition;
+
+    pxr::GfVec3d newEcefRotation;
+    globeAnchorAPI.GetRotationAttr().Get(&newEcefRotation);
+    _valueCache.ecefRotation = newEcefRotation;
+
+    pxr::GfVec3d newEcefScale;
+    globeAnchorAPI.GetScaleAttr().Get(&newEcefScale);
+    _valueCache.ecefScale = newEcefScale;
+
+    auto maybeNewTransform = UsdUtil::getCesiumTransformOpValueForPathIfExists(_anchorPrimPath);
+    if (maybeNewTransform.has_value()) {
+        _valueCache.transformation = maybeNewTransform.value();
+    }
+}
+
+OmniGlobeAnchor::OmniGlobeAnchor(pxr::SdfPath anchorPrimPath, const glm::dmat4 anchorToFixed)
+    : _anchorPrimPath{std::move(anchorPrimPath)} {
+    _anchor = std::make_shared<CesiumGeospatial::GlobeAnchor>(anchorToFixed);
```

Comment:
I ran into troubles with MSVC for this a while ago. Going to skip this for now since it's just style. I believe a proper fix for this is we have to use `explicit` but that of course comes with caveats.